packages/harmony/src/components/button/types.ts

```diff
@@ -131,3 +132,79 @@ export type PlainButtonProps = {
    */
   size?: PlainButtonSize
 } & Omit<BaseButtonProps, 'styles'>
+
+export enum FilterButtonSize {
+  DEFAULT = 'default',
+  SMALL = 'small'
+}
+
+export enum FilterButtonType {
+  /**
+   * The container is filled with solid color after selection
+   */
+  FILL_CONTAINER = 'fillContainer',
+
+  /**
+   * The label is used as the selected value
+   */
+  REPLACE_LABEL = 'replaceLabel'
+}
+
+export type FilterButtonOption = {
+  label: string
+  icon?: IconComponent
+}
+
+export type FilterButtonProps = {
+  /**
+   * Selection options
+   * e.g. { label: 'Option A', icon: IconRadar }
+   */
+  options: FilterButtonOption[]
+
+  /**
+   * The text that appears on the button component.
+   * If no label is provided, a different Icon can be specified
+   * to contextually inform users.
+   */
+  label?: string
```

Comment:
if no label provided, probably need to allow aria-label for button + listbox?

Reply:
nice